src/libpriv/rpmostree-jigdo-client.c

```diff
@@ -0,0 +1,263 @@
+    return FALSE;
+  { g_autofree guint8*csum = NULL;
+    if (!ostree_repo_write_metadata (repo, OSTREE_OBJECT_TYPE_COMMIT,
+                                     checksum, commit, &csum,
+                                     cancellable, error))
+      return FALSE;
+  }
+
+  if (!rpmostree_jigdo_assembler_write_new_objects (jigdo, repo, cancellable, error))
+    return FALSE;
+
+  if (!ostree_repo_commit_transaction (repo, NULL, cancellable, error))
+    return FALSE;
+  txn.initialized = FALSE;
+
+  /* And now, process the jigdo set */
+  if (!rpmostree_context_set_packages (self, pkgs_required, cancellable, error))
+    return FALSE;
+
+  /* See what packages we need to import, print their size. TODO clarify between
+   * download/import.
+   */
+  g_autoptr(GHashTable) pkgset_to_import = g_hash_table_new (NULL, NULL);
+  { g_autoptr(GPtrArray) pkgs_to_import = rpmostree_context_get_packages_to_import (self);
+    guint64 dlsize = 0;
+    for (guint i = 0; i < pkgs_to_import->len; i++)
+      {
+        DnfPackage *pkg = pkgs_to_import->pdata[i];
+        dlsize += dnf_package_get_size (pkg);
+        g_hash_table_add (pkgset_to_import, pkg);
+      }
+    g_autofree char *dlsize_fmt = g_format_size (dlsize);
+    rpmostree_output_message ("%u packages to import, download size: %s", pkgs_to_import->len, dlsize_fmt);
+  }
+
+  /* Parse the xattr data in the jigdoRPM */
+  g_autoptr(GHashTable) pkg_to_xattrs = g_hash_table_new_full (NULL, NULL,
+                                                               (GDestroyNotify)g_object_unref,
+                                                               (GDestroyNotify)g_variant_unref);
+
+  for (guint i = 0; i < pkgs_required->len; i++)
+    {
+      DnfPackage *pkg = pkgs_required->pdata[i];
+      const gboolean should_import = g_hash_table_contains (pkgset_to_import, pkg);
+      g_autoptr(GVariant) objid_to_xattrs = NULL;
+      if (!rpmostree_jigdo_assembler_next_xattrs (jigdo, &objid_to_xattrs, cancellable, error))
+        return FALSE;
+      if (!objid_to_xattrs)
+        return glnx_throw (error, "missing xattr entry: %s", dnf_package_get_name (pkg));
+      if (!should_import)
+        continue;
+      g_hash_table_insert (pkg_to_xattrs, g_object_ref (pkg), g_steal_pointer (&objid_to_xattrs));
+    }
+
+  /* Start the download and import, using the xattr data from the jigdoRPM */
+  if (!rpmostree_context_download (self, cancellable, error))
+    return FALSE;
+  g_autoptr(GVariant) xattr_table = rpmostree_jigdo_assembler_get_xattr_table (jigdo);
+  if (!rpmostree_context_import_jigdo (self, xattr_table, pkg_to_xattrs,
+                                       cancellable, error))
+    return FALSE;
+
+  /* Last thing is to delete the partial marker, just like
+   * ostree_repo_pull_with_options().
+   */
+  if (!ostree_repo_mark_commit_partial (repo, checksum, FALSE, error))
+    return FALSE;
+
+  *out_changed = FALSE;
```

Comment:
This should be `TRUE`, no?

Reply:
`TRUE`!  Though it's not actually used at the moment, I wrote it at first and then only later realized that the sysroot upgrader logic compares the commits.  Maybe I'll drop it later.